src/DIPS.Xamarin.UI/Commands/AsyncCommand.cs

```diff
@@ -114,25 +118,31 @@ public bool CanExecute(object parameter)
         }
 
         /// <inheritdoc />
-        public Task ExecuteAsync(T value) => m_execute(value);
-
-#pragma warning disable RECS0165
-        /// <inheritdoc />
-        public async void Execute(object parameter)
+        public Task ExecuteAsync(T value)
         {
+            if (value == null) return Task.CompletedTask;
             try
             {
-                if(!(parameter is T value) || !m_canExecute(value))
+                if (!(value is T actualValue) || !m_canExecute(actualValue))
                 {
-                    return;
+                    return Task.CompletedTask;
                 }
 
-                await ExecuteAsync(value);
+                return m_execute(actualValue);
```

Comment:
await and remove all return Task.CompletedTask
If we return a task, the stacktrace would be buggy. Hence we need to await all tasks.